torchaudio/datasets/librispeech.py

```diff
@@ -102,22 +101,45 @@ def __init__(
         folder_in_archive: str = FOLDER_IN_ARCHIVE,
         download: bool = False,
     ) -> None:
+        self._url = url
         if url not in _DATA_SUBSETS:
             raise ValueError(f"Invalid url '{url}' given; please provide one of {_DATA_SUBSETS}.")
 
         root = os.fspath(root)
+        self._archive = os.path.join(root, folder_in_archive)
         self._path = os.path.join(root, folder_in_archive, url)
 
         if not os.path.isdir(self._path):
             if download:
-                download_librispeech(root, url)
+                _download_librispeech(root, url)
             else:
                 raise RuntimeError(
                     f"Dataset not found at {self._path}. Please set `download=True` to download the dataset."
                 )
 
         self._walker = sorted(str(p.stem) for p in Path(self._path).glob("*/*/*" + self._ext_audio))
 
+    def get_metadata(self, n: int) -> Tuple[Tensor, int, str, int, int, int]:
```

Comment:
The type annotation is incorrect. The fix: #3757. @carolineechen @mthrok @nateanl